diff --git a/ATClangObfuscatorTest/ATClangObfuscatorTest/Classes/ATAnimal.h b/ATClangObfuscatorTest/ATClangObfuscatorTest/Classes/ATAnimal.h
index f897ff6..81423d9 100644
--- a/ATClangObfuscatorTest/ATClangObfuscatorTest/Classes/ATAnimal.h
+++ b/ATClangObfuscatorTest/ATClangObfuscatorTest/Classes/ATAnimal.h
@@ -34,16 +34,16 @@ NS_ASSUME_NONNULL_BEGIN
 - (void)setArea:(NSString *)area;
 
 /// should be obfuscated
-- (void)sleep;
+- (void)obfuscate_sleep;
 
 /// should be obfuscated
-- (void)sleepNow:(BOOL)now duration:(NSInteger)duration;
+- (void)obfuscate_sleep_now:(BOOL)now duration:(NSInteger)duration;
 
 /// should be obfuscated
-- (BOOL)sleepAtNight;
+- (BOOL)obfuscate_sleep_at_night;
 
 /// should be obfuscated
-+ (BOOL)isAnyInstanceLive;
++ (BOOL)obfuscate_is_any_instance_live;
 
 @end
 
diff --git a/ATClangObfuscatorTest/ATClangObfuscatorTest/Classes/ATAnimal.m b/ATClangObfuscatorTest/ATClangObfuscatorTest/Classes/ATAnimal.m
index dd819d8..1e388ba 100644
--- a/ATClangObfuscatorTest/ATClangObfuscatorTest/Classes/ATAnimal.m
+++ b/ATClangObfuscatorTest/ATClangObfuscatorTest/Classes/ATAnimal.m
@@ -19,7 +19,7 @@ static int kATAnimalLiveCount = 0;
 @property (nonatomic, assign) int privatePC;
 
 /// should be obfuscated
-- (void)privateMethod;
+- (void)obfuscate_private_method;
 
 @end
 
@@ -85,20 +85,20 @@ static int kATAnimalLiveCount = 0;
 }
 
 /// should be obfuscated
-- (void)sleep
+- (void)obfuscate_sleep
 {
     /// should be obfuscated
-    [self sleepNow:YES duration:100];
+    [self obfuscate_sleep_now:YES duration:100];
 }
 
 /// should be obfuscated
-- (void)sleepNow:(BOOL)now duration:(NSInteger)duration
+- (void)obfuscate_sleep_now:(BOOL)now duration:(NSInteger)duration
 {
     NSLog(@"do nothing");
 }
 
 /// should be obfuscated
-- (BOOL)sleepAtNight
+- (BOOL)obfuscate_sleep_at_night
 {
     return YES;
 }
@@ -116,19 +116,19 @@ static int kATAnimalLiveCount = 0;
 }
 
 /// should be obfuscated
-- (void)privateMethod
+- (void)obfuscate_private_method
 {
     NSLog(@"do nothing");
 }
 
 /// should be obfuscated
-- (void)privateMethodEx
+- (void)obfuscate_private_method_ex
 {
     NSLog(@"do nothing");
 }
 
 /// should be obfuscated
-+ (BOOL)isAnyInstanceLive
++ (BOOL)obfuscate_is_any_instance_live
 {
     return kATAnimalLiveCount > 0;
 }
diff --git a/ATClangObfuscatorTest/ATClangObfuscatorTest/Classes/ATBird.h b/ATClangObfuscatorTest/ATClangObfuscatorTest/Classes/ATBird.h
index 0ae45df..155f297 100644
--- a/ATClangObfuscatorTest/ATClangObfuscatorTest/Classes/ATBird.h
+++ b/ATClangObfuscatorTest/ATClangObfuscatorTest/Classes/ATBird.h
@@ -18,13 +18,13 @@ NS_ASSUME_NONNULL_BEGIN
 @property (nonatomic, assign) BOOL full;
 
 /// should be obfuscated
-- (void)flyEx;
+- (void)obfuscate_fly_ex;
 
 /// should be obfuscated
-- (int)count;
+- (int)obfuscate_count;
 
 /// should be obfuscated
-- (void)setCount:(int)count;
+- (void)setObfuscate_count:(int)count;
 
 @end
 
diff --git a/ATClangObfuscatorTest/ATClangObfuscatorTest/Classes/ATBird.m b/ATClangObfuscatorTest/ATClangObfuscatorTest/Classes/ATBird.m
index d7580bc..54d44ec 100644
--- a/ATClangObfuscatorTest/ATClangObfuscatorTest/Classes/ATBird.m
+++ b/ATClangObfuscatorTest/ATClangObfuscatorTest/Classes/ATBird.m
@@ -11,32 +11,32 @@
 @implementation ATBird
 
 /// should be obfuscated
-- (BOOL)sleepAtNight
+- (BOOL)obfuscate_sleep_at_night
 {
     return YES;
 }
 
 /// should be obfuscated
-- (void)fly
+- (void)obfuscate_fly
 {
     /// should be obfuscated
-    [self flyEx];
+    [self obfuscate_fly_ex];
 }
 
 /// should be obfuscated
-- (void)flyEx
+- (void)obfuscate_fly_ex
 {
     NSLog(@"do nothing");
 }
 
 /// should be obfuscated
-- (int)count
+- (int)obfuscate_count
 {
     return 0;
 }
 
 /// should be obfuscated
-- (void)setCount:(int)count
+- (void)setObfuscate_count:(int)count
 {
     NSLog(@"do nothing");
 }
diff --git a/ATClangObfuscatorTest/ATClangObfuscatorTest/Classes/ATFish.h b/ATClangObfuscatorTest/ATClangObfuscatorTest/Classes/ATFish.h
index 3dfe5f6..f87eb72 100644
--- a/ATClangObfuscatorTest/ATClangObfuscatorTest/Classes/ATFish.h
+++ b/ATClangObfuscatorTest/ATClangObfuscatorTest/Classes/ATFish.h
@@ -15,10 +15,10 @@ NS_ASSUME_NONNULL_BEGIN
 @interface ATFish : ATAnimal<IATFish>
 
 /// should be obfuscated
-- (void)swimEx;
+- (void)obfuscate_swim_ex;
 
 /// should be obfuscated
-+ (ATFish *)createInstance;
++ (ATFish *)obfuscate_create_instance;
 
 @end
 
diff --git a/ATClangObfuscatorTest/ATClangObfuscatorTest/Classes/ATFish.m b/ATClangObfuscatorTest/ATClangObfuscatorTest/Classes/ATFish.m
index 3805657..2379b68 100644
--- a/ATClangObfuscatorTest/ATClangObfuscatorTest/Classes/ATFish.m
+++ b/ATClangObfuscatorTest/ATClangObfuscatorTest/Classes/ATFish.m
@@ -11,26 +11,26 @@
 @implementation ATFish
 
 /// should be obfuscated
-- (BOOL)sleepAtNight
+- (BOOL)obfuscate_sleep_at_night
 {
     return YES;
 }
 
 /// should be obfuscated
-- (void)swim
+- (void)obfuscate_swim
 {
     /// should be obfuscated
-    [self swimEx];
+    [self obfuscate_swim_ex];
 }
 
 /// should be obfuscated
-- (void)swimEx
+- (void)obfuscate_swim_ex
 {
     NSLog(@"do nothing");
 }
 
 /// should be obfuscated
-+ (ATFish *)createInstance
++ (ATFish *)obfuscate_create_instance
 {
     return [[ATFish alloc] init];
 }
diff --git a/ATClangObfuscatorTest/ATClangObfuscatorTest/Classes/ATMacro.h b/ATClangObfuscatorTest/ATClangObfuscatorTest/Classes/ATMacro.h
index 5a8ea5a..90eb837 100644
--- a/ATClangObfuscatorTest/ATClangObfuscatorTest/Classes/ATMacro.h
+++ b/ATClangObfuscatorTest/ATClangObfuscatorTest/Classes/ATMacro.h
@@ -14,10 +14,10 @@
 /// sharedObject should be obfuscated
 
 #define AT_DECLARE_SINGLETON \
-+ (instancetype)sharedObject;
++ (instancetype)obfuscate_shared_object;
 
 #define AT_IMPLEMENT_SINGLETON(atType) \
-+ (instancetype)sharedObject { \
++ (instancetype)obfuscate_shared_object { \
     static dispatch_once_t __once; \
     static atType *__instance = nil; \
     dispatch_once(&__once, ^{ \
diff --git a/ATClangObfuscatorTest/ATClangObfuscatorTest/Classes/ATMonster+Action.h b/ATClangObfuscatorTest/ATClangObfuscatorTest/Classes/ATMonster+Action.h
index e4d0538..0408fc9 100644
--- a/ATClangObfuscatorTest/ATClangObfuscatorTest/Classes/ATMonster+Action.h
+++ b/ATClangObfuscatorTest/ATClangObfuscatorTest/Classes/ATMonster+Action.h
@@ -16,10 +16,10 @@ NS_ASSUME_NONNULL_BEGIN
 @property (nonatomic, assign) BOOL alive;
 
 /// should be obfuscated
-- (void)doAction;
+- (void)obfuscate_do_action;
 
 /// should be obfuscated
-+ (void)doClassAction;
++ (void)obfuscate_do_class_action;
 
 @end
 
@@ -29,7 +29,7 @@ NS_ASSUME_NONNULL_BEGIN
 - (BOOL)dead;
 
 /// should be obfuscated
-- (void)doActionEx;
+- (void)obfuscate_do_action_ex;
 
 /// property should NOT be obfuscated
 - (NSString *)otherName;
diff --git a/ATClangObfuscatorTest/ATClangObfuscatorTest/Classes/ATMonster+Action.m b/ATClangObfuscatorTest/ATClangObfuscatorTest/Classes/ATMonster+Action.m
index 9b97457..59fdac0 100644
--- a/ATClangObfuscatorTest/ATClangObfuscatorTest/Classes/ATMonster+Action.m
+++ b/ATClangObfuscatorTest/ATClangObfuscatorTest/Classes/ATMonster+Action.m
@@ -12,7 +12,7 @@
 @protocol IATActionBase <NSObject>
 
 /// should be obfuscated
-- (NSString *)base;
+- (NSString *)obfuscate_base;
 
 @end
 
@@ -23,7 +23,7 @@
 @implementation ATMonster (ActionPrivate)
 
 /// should be obfuscated
-- (NSString *)base;
+- (NSString *)obfuscate_base;
 {
     return nil;
 }
@@ -33,13 +33,13 @@
 @implementation ATMonster (Action)
 
 /// should be obfuscated
-- (void)doAction
+- (void)obfuscate_do_action
 {
     NSLog(@"do nothing");
 }
 
 /// should be obfuscated
-+ (void)doClassAction
++ (void)obfuscate_do_class_action
 {
     NSLog(@"do nothing");
 }
@@ -73,7 +73,7 @@
 }
 
 /// should be obfuscated
-- (void)doActionEx
+- (void)obfuscate_do_action_ex
 {
     NSLog(@"do nothing");
 }
diff --git a/ATClangObfuscatorTest/ATClangObfuscatorTest/Classes/ATMonster.h b/ATClangObfuscatorTest/ATClangObfuscatorTest/Classes/ATMonster.h
index 11f3fa9..ab84e0b 100644
--- a/ATClangObfuscatorTest/ATClangObfuscatorTest/Classes/ATMonster.h
+++ b/ATClangObfuscatorTest/ATClangObfuscatorTest/Classes/ATMonster.h
@@ -18,10 +18,10 @@ NS_ASSUME_NONNULL_BEGIN
 AT_EXTERN_NOTIFI(ATMonsterCreate);
 
 /// should be obfuscated
-#define ATMonsterCenterObj [ATMonsterCenter sharedObject]
+#define ATMonsterCenterObj [ATMonsterCenter obfuscate_shared_object]
 
 /// should be obfuscated
-#define ATMonsterCenterObjCreate [ATMonsterCenterObj create];
+#define ATMonsterCenterObjCreate [ATMonsterCenterObj obfuscate_create];
 
 @class ATMonster;
 
@@ -48,7 +48,7 @@ AT_EXTERN_NOTIFI(ATMonsterCreate);
 AT_DECLARE_SINGLETON;
 
 /// should be obfuscated
-- (ATMonster *)create;
+- (ATMonster *)obfuscate_create;
 
 /// getter/setter should NOT be obfuscated
 - (UIView *)currentMonsterView;
diff --git a/ATClangObfuscatorTest/ATClangObfuscatorTest/Classes/ATMonster.m b/ATClangObfuscatorTest/ATClangObfuscatorTest/Classes/ATMonster.m
index 31e57e2..f8a025d 100644
--- a/ATClangObfuscatorTest/ATClangObfuscatorTest/Classes/ATMonster.m
+++ b/ATClangObfuscatorTest/ATClangObfuscatorTest/Classes/ATMonster.m
@@ -83,33 +83,33 @@ AT_DECLARE_NOTIFI(ATMonsterCreate);
 @implementation ATMonster
 
 /// should be obfuscated
-- (void)swim
+- (void)obfuscate_swim
 {
     NSLog(@"do nothing");
 }
 
 /// should be obfuscated
-- (void)fly
+- (void)obfuscate_fly
 {
     NSLog(@"do nothing");
 }
 
 /// should be obfuscated
-- (void)scare
+- (void)obfuscate_scare
 {
     /// should be obfuscated
-    [self privateScare];
+    [self obfuscate_private_scare];
 }
 
 /// should be obfuscated
-- (void)privateScare
+- (void)obfuscate_private_scare
 {
     /// should be obfuscated
-    [self privateScareEx];
+    [self obfuscate_private_scare_ex];
 }
 
 /// should be obfuscated
-- (void)privateScareEx
+- (void)obfuscate_private_scare_ex
 {
     NSLog(@"do nothing");
 }
@@ -118,7 +118,7 @@ AT_DECLARE_NOTIFI(ATMonsterCreate);
 - (void)run
 {
     /// should be obfuscated
-    [self privateScareEx];
+    [self obfuscate_private_scare_ex];
 }
 
 @end
@@ -127,7 +127,7 @@ AT_DECLARE_NOTIFI(ATMonsterCreate);
 @implementation ATMonsterEx
 
 /// should be obfuscated
-- (void)scare
+- (void)obfuscate_scare
 {
     NSLog(@"do nothing");
 }
@@ -145,7 +145,7 @@ AT_DECLARE_NOTIFI(ATMonsterCreate);
 }
 
 /// should be obfuscated
-- (void)scareEx
+- (void)obfuscate_scare_ex
 {
     NSLog(@"do nothing");
 }
@@ -170,7 +170,7 @@ AT_DECLARE_NOTIFI(ATMonsterCreate);
 @implementation ATMonsterEx (OnlyImp)
 
 /// should be obfuscated
-- (void)onlyImp
+- (void)obfuscate_only_imp
 {
     NSLog(@"do nothing");
 }
@@ -190,20 +190,20 @@ AT_DECLARE_NOTIFI(ATMonsterCreate);
 AT_IMPLEMENT_SINGLETON(ATMonsterCenter);
 
 /// should be obfuscated
-- (ATMonster *)create
+- (ATMonster *)obfuscate_create
 {
     AT_POST_NOTIFY(ATMonsterCreate);
     return [ATMonster new];
 }
 
 /// should be obfuscated
-- (ATMonsterEx *)createEx
+- (ATMonsterEx *)obfuscate_create_ex
 {
     /// new should NOT be obfuscated
     ATMonsterEx *ex = [ATMonsterEx new];
     
     /// should be obfuscated
-    [ex onlyImp];
+    [ex obfuscate_only_imp];
     
     /// should NOT be obfuscated
     [ex level];
diff --git a/ATClangObfuscatorTest/ATClangObfuscatorTest/Classes/ATTest.h b/ATClangObfuscatorTest/ATClangObfuscatorTest/Classes/ATTest.h
index 7942045..dab6fe0 100644
--- a/ATClangObfuscatorTest/ATClangObfuscatorTest/Classes/ATTest.h
+++ b/ATClangObfuscatorTest/ATClangObfuscatorTest/Classes/ATTest.h
@@ -21,10 +21,10 @@ AT_DECLARE_SINGLETON;
 AT_DECLARE_CREATE_OBJ;
 
 /// should be obfuscated
-- (void)test;
+- (void)obfuscate_test;
 
 /// should be obfuscated
-- (void)setMonster:(ATMonster *)monster;
+- (void)setObfuscate_monster:(ATMonster *)monster;
 
 @end
 
diff --git a/ATClangObfuscatorTest/ATClangObfuscatorTest/Classes/ATTest.m b/ATClangObfuscatorTest/ATClangObfuscatorTest/Classes/ATTest.m
index 4cbad09..f3edf22 100644
--- a/ATClangObfuscatorTest/ATClangObfuscatorTest/Classes/ATTest.m
+++ b/ATClangObfuscatorTest/ATClangObfuscatorTest/Classes/ATTest.m
@@ -28,22 +28,22 @@ AT_IMPLEMENT_CREATE_OBJ(ATTest);
 }
 
 /// should be obfuscated
-- (void)test
+- (void)obfuscate_test
 {
     /// should be obfuscated
-    ATFish *fish = [ATFish createInstance];
+    ATFish *fish = [ATFish obfuscate_create_instance];
     
     /// should be obfuscated
-    [fish swim];
+    [fish obfuscate_swim];
     
     /// should be obfuscated
-    [fish swimEx];
+    [fish obfuscate_swim_ex];
     
     /// new should NOT be obfuscated
     id<IATBird> iBird = [ATBird new];
     
     /// should be obfuscated
-    [iBird fly];
+    [iBird obfuscate_fly];
     
     /// new should NOT be obfuscated
     ATBird *bird = [ATBird new];
@@ -61,31 +61,31 @@ AT_IMPLEMENT_CREATE_OBJ(ATTest);
     bird.full = full;
     
     /// not property should be obfuscated
-    int count = bird.count;
+    int count = bird.obfuscate_count;
     
     /// not property should be obfuscated
-    bird.count = count;
+    bird.obfuscate_count = count;
 }
 
 /// should be obfuscated
-- (void)setMonster:(ATMonster *)monster
+- (void)setObfuscate_monster:(ATMonster *)monster
 {
     /// should be obfuscated
-    [monster sleep];
+    [monster obfuscate_sleep];
     
     /// should be obfuscated
-    [monster doAction];
+    [monster obfuscate_do_action];
     
     /// run should NOT be obfuscated
     [monster performSelector:@selector(run)];
     
     /// should be obfuscated
-    [ATMonster doClassAction];
+    [ATMonster obfuscate_do_class_action];
     
     id tmp = monster;
     
     /// should be obfuscated
-    [tmp sleep];
+    [tmp obfuscate_sleep];
     
     /// run should NOT be obfuscated
     [tmp run];
@@ -94,13 +94,13 @@ AT_IMPLEMENT_CREATE_OBJ(ATTest);
     id<IATMonster> monsterEx = [ATMonsterEx new];
     
     /// should be obfuscated
-    [monsterEx scare];
+    [monsterEx obfuscate_scare];
     
     /// should be obfuscated
-    ATTest *newObj = [ATTest createObj];
+    ATTest *newObj = [ATTest obfuscate_create_obj];
     
     /// should be obfuscated
-    [newObj test];
+    [newObj obfuscate_test];
     
     /// onATMonsterCreate is written in ScratchSpace should NOT be obfuscated
     [self onATMonsterCreate:nil];
@@ -109,7 +109,7 @@ AT_IMPLEMENT_CREATE_OBJ(ATTest);
 AT_HANDLER_NOTIFY_DEFAULT_SELECTOR_NAME(ATMonsterCreate)
 {
     /// should be obfuscated
-    [ATMonsterCenterObj create];
+    [ATMonsterCenterObj obfuscate_create];
     
     ATMonsterCenterObjCreate;
     
diff --git a/ATClangObfuscatorTest/ATClangObfuscatorTest/Classes/Define/ATDefine.h b/ATClangObfuscatorTest/ATClangObfuscatorTest/Classes/Define/ATDefine.h
index e75a964..dbe9ed5 100644
--- a/ATClangObfuscatorTest/ATClangObfuscatorTest/Classes/Define/ATDefine.h
+++ b/ATClangObfuscatorTest/ATClangObfuscatorTest/Classes/Define/ATDefine.h
@@ -12,10 +12,10 @@
 /// createObj should be obfuscated
 
 #define AT_DECLARE_CREATE_OBJ \
-+ (instancetype)createObj;
++ (instancetype)obfuscate_create_obj;
 
 #define AT_IMPLEMENT_CREATE_OBJ(atType) \
-+ (instancetype)createObj { \
++ (instancetype)obfuscate_create_obj { \
     return [[atType alloc] init]; \
 }
 
diff --git a/ATClangObfuscatorTest/ATClangObfuscatorTest/Classes/IATBird.h b/ATClangObfuscatorTest/ATClangObfuscatorTest/Classes/IATBird.h
index edb48db..e46afb3 100644
--- a/ATClangObfuscatorTest/ATClangObfuscatorTest/Classes/IATBird.h
+++ b/ATClangObfuscatorTest/ATClangObfuscatorTest/Classes/IATBird.h
@@ -13,7 +13,7 @@ NS_ASSUME_NONNULL_BEGIN
 @protocol IATBird <NSObject>
 
 /// should be obfuscated
-- (void)fly;
+- (void)obfuscate_fly;
 
 @end
 
diff --git a/ATClangObfuscatorTest/ATClangObfuscatorTest/Classes/IATFish.h b/ATClangObfuscatorTest/ATClangObfuscatorTest/Classes/IATFish.h
index 3224c42..82d9b9e 100644
--- a/ATClangObfuscatorTest/ATClangObfuscatorTest/Classes/IATFish.h
+++ b/ATClangObfuscatorTest/ATClangObfuscatorTest/Classes/IATFish.h
@@ -13,7 +13,7 @@ NS_ASSUME_NONNULL_BEGIN
 @protocol IATFish <NSObject>
 
 /// should be obfuscated
-- (void)swim;
+- (void)obfuscate_swim;
 
 @end
 
diff --git a/ATClangObfuscatorTest/ATClangObfuscatorTest/Classes/IATMonster.h b/ATClangObfuscatorTest/ATClangObfuscatorTest/Classes/IATMonster.h
index dc93605..26acb98 100644
--- a/ATClangObfuscatorTest/ATClangObfuscatorTest/Classes/IATMonster.h
+++ b/ATClangObfuscatorTest/ATClangObfuscatorTest/Classes/IATMonster.h
@@ -13,7 +13,7 @@ NS_ASSUME_NONNULL_BEGIN
 @protocol IATMonster <NSObject>
 
 /// should be obfuscated
-- (void)scare;
+- (void)obfuscate_scare;
 
 /// should NOT be obfuscated : ATMonster conforms to IATMonster and it has property named amIBig
 - (void)amIBig;
@@ -32,13 +32,13 @@ NS_ASSUME_NONNULL_BEGIN
 @property (nonatomic, assign) BOOL hasEye;
 
 /// should be obfuscated
-- (void)scare;
+- (void)obfuscate_scare;
 
 /// should NOT be obfuscated : ATMonster conforms to IATMonster and it has property named amIBig
 - (void)amIBig;
 
 /// should be obfuscated
-- (void)scareEx;
+- (void)obfuscate_scare_ex;
 
 @end
 
diff --git a/ATClangObfuscatorTest/ATClangObfuscatorTest/Classes/Other/ATOther.h b/ATClangObfuscatorTest/ATClangObfuscatorTest/Classes/Other/ATOther.h
index 34767a2..0289f7a 100644
--- a/ATClangObfuscatorTest/ATClangObfuscatorTest/Classes/Other/ATOther.h
+++ b/ATClangObfuscatorTest/ATClangObfuscatorTest/Classes/Other/ATOther.h
@@ -17,7 +17,7 @@ NS_ASSUME_NONNULL_BEGIN
 @interface NSObject (ATOther)
 
 /// should be obfuscated
-- (void)publicMethod;
+- (void)obfuscate_public_method;
 
 @end
 
diff --git a/ATClangObfuscatorTest/ATClangObfuscatorTest/Classes/Other/ATOther.m b/ATClangObfuscatorTest/ATClangObfuscatorTest/Classes/Other/ATOther.m
index cfd6e5c..aac2fd9 100644
--- a/ATClangObfuscatorTest/ATClangObfuscatorTest/Classes/Other/ATOther.m
+++ b/ATClangObfuscatorTest/ATClangObfuscatorTest/Classes/Other/ATOther.m
@@ -17,13 +17,13 @@
     self = [super init];
     if (self) {
         /// should be obfuscated
-        [self other];
+        [self obfuscate_other];
     }
     return self;
 }
 
 /// should be obfuscated
-- (void)other
+- (void)obfuscate_other
 {
     /// sys method should NOT be obfuscated
     UIView *tmp = [[UIView alloc] initWithFrame:CGRectZero];
@@ -32,17 +32,17 @@
     [tmp sizeToFit];
     
     /// should be obfuscated
-    [self private];
+    [self obfuscate_private];
     
     /// sys method should NOT be obfuscated
     ATAnimal *animal = [ATAnimal new];
     
     /// should be obfuscated
-    [animal sleep];
+    [animal obfuscate_sleep];
 }
 
 /// should be obfuscated
-- (void)private
+- (void)obfuscate_private
 {
     ;;
 }
@@ -53,14 +53,14 @@
 @implementation NSObject (ATOther)
 
 /// should be obfuscated
-- (void)publicMethod
+- (void)obfuscate_public_method
 {
     /// should be obfuscated
-    [self privateMethod];
+    [self obfuscate_private_method];
 }
 
 /// should be obfuscated
-- (void)privateMethod
+- (void)obfuscate_private_method
 {
     ;;
 }
