diff --git a/ATClangObfuscatorTest/ATClangObfuscatorTest/Classes/ATAnimal.h b/ATClangObfuscatorTest/ATClangObfuscatorTest/Classes/ATAnimal.h
index f897ff6..b7517f0 100644
--- a/ATClangObfuscatorTest/ATClangObfuscatorTest/Classes/ATAnimal.h
+++ b/ATClangObfuscatorTest/ATClangObfuscatorTest/Classes/ATAnimal.h
@@ -34,16 +34,16 @@ NS_ASSUME_NONNULL_BEGIN
 - (void)setArea:(NSString *)area;
 
 /// should be obfuscated
-- (void)sleep;
+- (void)obfuscate_sleep;
 
 /// should be obfuscated
-- (void)sleepNow:(BOOL)now duration:(NSInteger)duration;
+- (void)obfuscate_sleepNow:(BOOL)now duration:(NSInteger)duration;
 
 /// should be obfuscated
-- (BOOL)sleepAtNight;
+- (BOOL)obfuscate_sleepAtNight;
 
 /// should be obfuscated
-+ (BOOL)isAnyInstanceLive;
++ (BOOL)obfuscate_isAnyInstanceLive;
 
 @end
 
diff --git a/ATClangObfuscatorTest/ATClangObfuscatorTest/Classes/ATAnimal.m b/ATClangObfuscatorTest/ATClangObfuscatorTest/Classes/ATAnimal.m
index dd819d8..06af6ff 100644
--- a/ATClangObfuscatorTest/ATClangObfuscatorTest/Classes/ATAnimal.m
+++ b/ATClangObfuscatorTest/ATClangObfuscatorTest/Classes/ATAnimal.m
@@ -19,7 +19,7 @@ static int kATAnimalLiveCount = 0;
 @property (nonatomic, assign) int privatePC;
 
 /// should be obfuscated
-- (void)privateMethod;
+- (void)obfuscate_privateMethod;
 
 @end
 
@@ -85,20 +85,20 @@ static int kATAnimalLiveCount = 0;
 }
 
 /// should be obfuscated
-- (void)sleep
+- (void)obfuscate_sleep
 {
     /// should be obfuscated
-    [self sleepNow:YES duration:100];
+    [self obfuscate_sleepNow:YES duration:100];
 }
 
 /// should be obfuscated
-- (void)sleepNow:(BOOL)now duration:(NSInteger)duration
+- (void)obfuscate_sleepNow:(BOOL)now duration:(NSInteger)duration
 {
     NSLog(@"do nothing");
 }
 
 /// should be obfuscated
-- (BOOL)sleepAtNight
+- (BOOL)obfuscate_sleepAtNight
 {
     return YES;
 }
@@ -116,19 +116,19 @@ static int kATAnimalLiveCount = 0;
 }
 
 /// should be obfuscated
-- (void)privateMethod
+- (void)obfuscate_privateMethod
 {
     NSLog(@"do nothing");
 }
 
 /// should be obfuscated
-- (void)privateMethodEx
+- (void)obfuscate_privateMethodEx
 {
     NSLog(@"do nothing");
 }
 
 /// should be obfuscated
-+ (BOOL)isAnyInstanceLive
++ (BOOL)obfuscate_isAnyInstanceLive
 {
     return kATAnimalLiveCount > 0;
 }
diff --git a/ATClangObfuscatorTest/ATClangObfuscatorTest/Classes/ATBird.h b/ATClangObfuscatorTest/ATClangObfuscatorTest/Classes/ATBird.h
index 0ae45df..a06ef5e 100644
--- a/ATClangObfuscatorTest/ATClangObfuscatorTest/Classes/ATBird.h
+++ b/ATClangObfuscatorTest/ATClangObfuscatorTest/Classes/ATBird.h
@@ -18,13 +18,13 @@ NS_ASSUME_NONNULL_BEGIN
 @property (nonatomic, assign) BOOL full;
 
 /// should be obfuscated
-- (void)flyEx;
+- (void)obfuscate_flyEx;
 
 /// should be obfuscated
-- (int)count;
+- (int)obfuscate_count;
 
 /// should be obfuscated
-- (void)setCount:(int)count;
+- (void)setObfuscate_count:(int)count;
 
 @end
 
diff --git a/ATClangObfuscatorTest/ATClangObfuscatorTest/Classes/ATBird.m b/ATClangObfuscatorTest/ATClangObfuscatorTest/Classes/ATBird.m
index d7580bc..c078f0a 100644
--- a/ATClangObfuscatorTest/ATClangObfuscatorTest/Classes/ATBird.m
+++ b/ATClangObfuscatorTest/ATClangObfuscatorTest/Classes/ATBird.m
@@ -11,32 +11,32 @@
 @implementation ATBird
 
 /// should be obfuscated
-- (BOOL)sleepAtNight
+- (BOOL)obfuscate_sleepAtNight
 {
     return YES;
 }
 
 /// should be obfuscated
-- (void)fly
+- (void)obfuscate_fly
 {
     /// should be obfuscated
-    [self flyEx];
+    [self obfuscate_flyEx];
 }
 
 /// should be obfuscated
-- (void)flyEx
+- (void)obfuscate_flyEx
 {
     NSLog(@"do nothing");
 }
 
 /// should be obfuscated
-- (int)count
+- (int)obfuscate_count
 {
     return 0;
 }
 
 /// should be obfuscated
-- (void)setCount:(int)count
+- (void)setObfuscate_count:(int)count
 {
     NSLog(@"do nothing");
 }
diff --git a/ATClangObfuscatorTest/ATClangObfuscatorTest/Classes/ATFish.h b/ATClangObfuscatorTest/ATClangObfuscatorTest/Classes/ATFish.h
index 3dfe5f6..3669591 100644
--- a/ATClangObfuscatorTest/ATClangObfuscatorTest/Classes/ATFish.h
+++ b/ATClangObfuscatorTest/ATClangObfuscatorTest/Classes/ATFish.h
@@ -15,10 +15,10 @@ NS_ASSUME_NONNULL_BEGIN
 @interface ATFish : ATAnimal<IATFish>
 
 /// should be obfuscated
-- (void)swimEx;
+- (void)obfuscate_swimEx;
 
 /// should be obfuscated
-+ (ATFish *)createInstance;
++ (ATFish *)obfuscate_createInstance;
 
 @end
 
diff --git a/ATClangObfuscatorTest/ATClangObfuscatorTest/Classes/ATFish.m b/ATClangObfuscatorTest/ATClangObfuscatorTest/Classes/ATFish.m
index 3805657..2800b85 100644
--- a/ATClangObfuscatorTest/ATClangObfuscatorTest/Classes/ATFish.m
+++ b/ATClangObfuscatorTest/ATClangObfuscatorTest/Classes/ATFish.m
@@ -11,26 +11,26 @@
 @implementation ATFish
 
 /// should be obfuscated
-- (BOOL)sleepAtNight
+- (BOOL)obfuscate_sleepAtNight
 {
     return YES;
 }
 
 /// should be obfuscated
-- (void)swim
+- (void)obfuscate_swim
 {
     /// should be obfuscated
-    [self swimEx];
+    [self obfuscate_swimEx];
 }
 
 /// should be obfuscated
-- (void)swimEx
+- (void)obfuscate_swimEx
 {
     NSLog(@"do nothing");
 }
 
 /// should be obfuscated
-+ (ATFish *)createInstance
++ (ATFish *)obfuscate_createInstance
 {
     return [[ATFish alloc] init];
 }
diff --git a/ATClangObfuscatorTest/ATClangObfuscatorTest/Classes/ATMacro.h b/ATClangObfuscatorTest/ATClangObfuscatorTest/Classes/ATMacro.h
index 5a8ea5a..639ff3a 100644
--- a/ATClangObfuscatorTest/ATClangObfuscatorTest/Classes/ATMacro.h
+++ b/ATClangObfuscatorTest/ATClangObfuscatorTest/Classes/ATMacro.h
@@ -14,10 +14,10 @@
 /// sharedObject should be obfuscated
 
 #define AT_DECLARE_SINGLETON \
-+ (instancetype)sharedObject;
++ (instancetype)obfuscate_sharedObject;
 
 #define AT_IMPLEMENT_SINGLETON(atType) \
-+ (instancetype)sharedObject { \
++ (instancetype)obfuscate_sharedObject { \
     static dispatch_once_t __once; \
     static atType *__instance = nil; \
     dispatch_once(&__once, ^{ \
diff --git a/ATClangObfuscatorTest/ATClangObfuscatorTest/Classes/ATMonster+Action.h b/ATClangObfuscatorTest/ATClangObfuscatorTest/Classes/ATMonster+Action.h
index 8eff7cc..aadda91 100644
--- a/ATClangObfuscatorTest/ATClangObfuscatorTest/Classes/ATMonster+Action.h
+++ b/ATClangObfuscatorTest/ATClangObfuscatorTest/Classes/ATMonster+Action.h
@@ -16,10 +16,10 @@ NS_ASSUME_NONNULL_BEGIN
 @property (nonatomic, assign) BOOL alive;
 
 /// should be obfuscated
-- (void)doAction;
+- (void)obfuscate_doAction;
 
 /// should be obfuscated
-+ (void)doClassAction;
++ (void)obfuscate_doClassAction;
 
 @end
 
@@ -29,7 +29,7 @@ NS_ASSUME_NONNULL_BEGIN
 - (BOOL)dead;
 
 /// should be obfuscated
-- (void)doActionEx;
+- (void)obfuscate_doActionEx;
 
 @end
 
diff --git a/ATClangObfuscatorTest/ATClangObfuscatorTest/Classes/ATMonster+Action.m b/ATClangObfuscatorTest/ATClangObfuscatorTest/Classes/ATMonster+Action.m
index e938a48..a8ae295 100644
--- a/ATClangObfuscatorTest/ATClangObfuscatorTest/Classes/ATMonster+Action.m
+++ b/ATClangObfuscatorTest/ATClangObfuscatorTest/Classes/ATMonster+Action.m
@@ -11,7 +11,7 @@
 @protocol IATActionBase <NSObject>
 
 /// should be obfuscated
-- (NSString *)base;
+- (NSString *)obfuscate_base;
 
 @end
 
@@ -22,7 +22,7 @@
 @implementation ATMonster (ActionPrivate)
 
 /// should be obfuscated
-- (NSString *)base;
+- (NSString *)obfuscate_base;
 {
     return nil;
 }
@@ -32,13 +32,13 @@
 @implementation ATMonster (Action)
 
 /// should be obfuscated
-- (void)doAction
+- (void)obfuscate_doAction
 {
     NSLog(@"do nothing");
 }
 
 /// should be obfuscated
-+ (void)doClassAction
++ (void)obfuscate_doClassAction
 {
     NSLog(@"do nothing");
 }
@@ -72,7 +72,7 @@
 }
 
 /// should be obfuscated
-- (void)doActionEx
+- (void)obfuscate_doActionEx
 {
     NSLog(@"do nothing");
 }
diff --git a/ATClangObfuscatorTest/ATClangObfuscatorTest/Classes/ATMonster.h b/ATClangObfuscatorTest/ATClangObfuscatorTest/Classes/ATMonster.h
index 11f3fa9..a50e98a 100644
--- a/ATClangObfuscatorTest/ATClangObfuscatorTest/Classes/ATMonster.h
+++ b/ATClangObfuscatorTest/ATClangObfuscatorTest/Classes/ATMonster.h
@@ -18,10 +18,10 @@ NS_ASSUME_NONNULL_BEGIN
 AT_EXTERN_NOTIFI(ATMonsterCreate);
 
 /// should be obfuscated
-#define ATMonsterCenterObj [ATMonsterCenter sharedObject]
+#define ATMonsterCenterObj [ATMonsterCenter obfuscate_sharedObject]
 
 /// should be obfuscated
-#define ATMonsterCenterObjCreate [ATMonsterCenterObj create];
+#define ATMonsterCenterObjCreate [ATMonsterCenterObj obfuscate_create];
 
 @class ATMonster;
 
@@ -48,7 +48,7 @@ AT_EXTERN_NOTIFI(ATMonsterCreate);
 AT_DECLARE_SINGLETON;
 
 /// should be obfuscated
-- (ATMonster *)create;
+- (ATMonster *)obfuscate_create;
 
 /// getter/setter should NOT be obfuscated
 - (UIView *)currentMonsterView;
diff --git a/ATClangObfuscatorTest/ATClangObfuscatorTest/Classes/ATMonster.m b/ATClangObfuscatorTest/ATClangObfuscatorTest/Classes/ATMonster.m
index 48a2ae8..c57eb43 100644
--- a/ATClangObfuscatorTest/ATClangObfuscatorTest/Classes/ATMonster.m
+++ b/ATClangObfuscatorTest/ATClangObfuscatorTest/Classes/ATMonster.m
@@ -83,33 +83,33 @@ AT_DECLARE_NOTIFI(ATMonsterCreate);
 @implementation ATMonster
 
 /// should be obfuscated
-- (void)swim
+- (void)obfuscate_swim
 {
     NSLog(@"do nothing");
 }
 
 /// should be obfuscated
-- (void)fly
+- (void)obfuscate_fly
 {
     NSLog(@"do nothing");
 }
 
 /// should be obfuscated
-- (void)scare
+- (void)obfuscate_scare
 {
     /// should be obfuscated
-    [self privateScare];
+    [self obfuscate_privateScare];
 }
 
 /// should be obfuscated
-- (void)privateScare
+- (void)obfuscate_privateScare
 {
     /// should be obfuscated
-    [self privateScareEx];
+    [self obfuscate_privateScareEx];
 }
 
 /// should be obfuscated
-- (void)privateScareEx
+- (void)obfuscate_privateScareEx
 {
     NSLog(@"do nothing");
 }
@@ -118,7 +118,7 @@ AT_DECLARE_NOTIFI(ATMonsterCreate);
 - (void)run
 {
     /// should be obfuscated
-    [self privateScareEx];
+    [self obfuscate_privateScareEx];
 }
 
 @end
@@ -127,7 +127,7 @@ AT_DECLARE_NOTIFI(ATMonsterCreate);
 @implementation ATMonsterEx
 
 /// should be obfuscated
-- (void)scare
+- (void)obfuscate_scare
 {
     NSLog(@"do nothing");
 }
@@ -145,7 +145,7 @@ AT_DECLARE_NOTIFI(ATMonsterCreate);
 }
 
 /// should be obfuscated
-- (void)scareEx
+- (void)obfuscate_scareEx
 {
     NSLog(@"do nothing");
 }
@@ -156,7 +156,7 @@ AT_DECLARE_NOTIFI(ATMonsterCreate);
 @implementation ATMonsterEx (OnlyImp)
 
 /// should be obfuscated
-- (void)onlyImp
+- (void)obfuscate_onlyImp
 {
     NSLog(@"do nothing");
 }
@@ -176,20 +176,20 @@ AT_DECLARE_NOTIFI(ATMonsterCreate);
 AT_IMPLEMENT_SINGLETON(ATMonsterCenter);
 
 /// should be obfuscated
-- (ATMonster *)create
+- (ATMonster *)obfuscate_create
 {
     AT_POST_NOTIFY(ATMonsterCreate);
     return [ATMonster new];
 }
 
 /// should be obfuscated
-- (ATMonsterEx *)createEx
+- (ATMonsterEx *)obfuscate_createEx
 {
     /// new should NOT be obfuscated
     ATMonsterEx *ex = [ATMonsterEx new];
     
     /// should be obfuscated
-    [ex onlyImp];
+    [ex obfuscate_onlyImp];
     
     return ex;
 }
diff --git a/ATClangObfuscatorTest/ATClangObfuscatorTest/Classes/ATTest.h b/ATClangObfuscatorTest/ATClangObfuscatorTest/Classes/ATTest.h
index 7942045..dab6fe0 100644
--- a/ATClangObfuscatorTest/ATClangObfuscatorTest/Classes/ATTest.h
+++ b/ATClangObfuscatorTest/ATClangObfuscatorTest/Classes/ATTest.h
@@ -21,10 +21,10 @@ AT_DECLARE_SINGLETON;
 AT_DECLARE_CREATE_OBJ;
 
 /// should be obfuscated
-- (void)test;
+- (void)obfuscate_test;
 
 /// should be obfuscated
-- (void)setMonster:(ATMonster *)monster;
+- (void)setObfuscate_monster:(ATMonster *)monster;
 
 @end
 
diff --git a/ATClangObfuscatorTest/ATClangObfuscatorTest/Classes/ATTest.m b/ATClangObfuscatorTest/ATClangObfuscatorTest/Classes/ATTest.m
index c2a2fca..6462401 100644
--- a/ATClangObfuscatorTest/ATClangObfuscatorTest/Classes/ATTest.m
+++ b/ATClangObfuscatorTest/ATClangObfuscatorTest/Classes/ATTest.m
@@ -28,22 +28,22 @@ AT_IMPLEMENT_CREATE_OBJ(ATTest);
 }
 
 /// should be obfuscated
-- (void)test
+- (void)obfuscate_test
 {
     /// should be obfuscated
-    ATFish *fish = [ATFish createInstance];
+    ATFish *fish = [ATFish obfuscate_createInstance];
     
     /// should be obfuscated
-    [fish swim];
+    [fish obfuscate_swim];
     
     /// should be obfuscated
-    [fish swimEx];
+    [fish obfuscate_swimEx];
     
     /// new should NOT be obfuscated
     id<IATBird> iBird = [ATBird new];
     
     /// should be obfuscated
-    [iBird fly];
+    [iBird obfuscate_fly];
     
     /// new should NOT be obfuscated
     ATBird *bird = [ATBird new];
@@ -61,31 +61,31 @@ AT_IMPLEMENT_CREATE_OBJ(ATTest);
     bird.full = full;
     
     /// not property should be obfuscated
-    int count = bird.count;
+    int count = bird.obfuscate_count;
     
     /// not property should be obfuscated
-    bird.count = count;
+    bird.obfuscate_count = count;
 }
 
 /// should be obfuscated
-- (void)setMonster:(ATMonster *)monster
+- (void)setObfuscate_monster:(ATMonster *)monster
 {
     /// should be obfuscated
-    [monster sleep];
+    [monster obfuscate_sleep];
     
     /// should be obfuscated
-    [monster doAction];
+    [monster obfuscate_doAction];
     
     /// run should NOT be obfuscated
     [monster performSelector:@selector(run)];
     
     /// should be obfuscated
-    [ATMonster doClassAction];
+    [ATMonster obfuscate_doClassAction];
     
     id tmp = monster;
     
     /// should be obfuscated
-    [tmp sleep];
+    [tmp obfuscate_sleep];
     
     /// run should NOT be obfuscated
     [tmp run];
@@ -94,19 +94,19 @@ AT_IMPLEMENT_CREATE_OBJ(ATTest);
     id<IATMonster> monsterEx = [ATMonsterEx new];
     
     /// should be obfuscated
-    [monsterEx scare];
+    [monsterEx obfuscate_scare];
     
     /// should be obfuscated
-    ATTest *newObj = [ATTest createObj];
+    ATTest *newObj = [ATTest obfuscate_createObj];
     
     /// should be obfuscated
-    [newObj test];
+    [newObj obfuscate_test];
 }
 
 AT_HANDLER_NOTIFY_DEFAULT_SELECTOR_NAME(ATMonsterCreate)
 {
     /// should be obfuscated
-    [ATMonsterCenterObj create];
+    [ATMonsterCenterObj obfuscate_create];
     
     ATMonsterCenterObjCreate;
 }
diff --git a/ATClangObfuscatorTest/ATClangObfuscatorTest/Classes/IATBird.h b/ATClangObfuscatorTest/ATClangObfuscatorTest/Classes/IATBird.h
index edb48db..e46afb3 100644
--- a/ATClangObfuscatorTest/ATClangObfuscatorTest/Classes/IATBird.h
+++ b/ATClangObfuscatorTest/ATClangObfuscatorTest/Classes/IATBird.h
@@ -13,7 +13,7 @@ NS_ASSUME_NONNULL_BEGIN
 @protocol IATBird <NSObject>
 
 /// should be obfuscated
-- (void)fly;
+- (void)obfuscate_fly;
 
 @end
 
diff --git a/ATClangObfuscatorTest/ATClangObfuscatorTest/Classes/IATFish.h b/ATClangObfuscatorTest/ATClangObfuscatorTest/Classes/IATFish.h
index 3224c42..82d9b9e 100644
--- a/ATClangObfuscatorTest/ATClangObfuscatorTest/Classes/IATFish.h
+++ b/ATClangObfuscatorTest/ATClangObfuscatorTest/Classes/IATFish.h
@@ -13,7 +13,7 @@ NS_ASSUME_NONNULL_BEGIN
 @protocol IATFish <NSObject>
 
 /// should be obfuscated
-- (void)swim;
+- (void)obfuscate_swim;
 
 @end
 
diff --git a/ATClangObfuscatorTest/ATClangObfuscatorTest/Classes/IATMonster.h b/ATClangObfuscatorTest/ATClangObfuscatorTest/Classes/IATMonster.h
index 79c959f..a5e2e45 100644
--- a/ATClangObfuscatorTest/ATClangObfuscatorTest/Classes/IATMonster.h
+++ b/ATClangObfuscatorTest/ATClangObfuscatorTest/Classes/IATMonster.h
@@ -13,7 +13,7 @@ NS_ASSUME_NONNULL_BEGIN
 @protocol IATMonster <NSObject>
 
 /// should be obfuscated
-- (void)scare;
+- (void)obfuscate_scare;
 
 /// should NOT be obfuscated : ATMonster conforms to IATMonster and it has property named amIBig
 - (void)amIBig;
@@ -26,13 +26,13 @@ NS_ASSUME_NONNULL_BEGIN
 @protocol IATMonsterEx <IATMonster>
 
 /// should be obfuscated
-- (void)scare;
+- (void)obfuscate_scare;
 
 /// should NOT be obfuscated : ATMonster conforms to IATMonster and it has property named amIBig
 - (void)amIBig;
 
 /// should be obfuscated
-- (void)scareEx;
+- (void)obfuscate_scareEx;
 
 @end
 
